confgenerator/logging_modify_fields.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confgenerator
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/filter"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+)
+
+type ModifyField struct {
+	// Source of value for this field
+	MoveFrom    string  `yaml:"move_from" validate:"omitempty,field,excluded_with=CopyFrom StaticValue"`
```

Comment:
Nope, because they're not required. None of the fields being specified is treated as `copy_from` the same metric; i.e. it reads the value from this field, applies the transformations, and writes back to the same field.

Reply:
To make sure I understand it fully, what is an example config that does not have any of `MoveFrom`, `CopyFrom` or `StaticValue`?

Reply:
Got it.